x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AgentConfigurationCreateEdit/ServicePage/ServicePage.tsx

```diff
@@ -0,0 +1,192 @@
+import { FormRowSelect } from './FormRowSelect';
+import { CancelButton } from './CancelButton';
+
+interface Props {
+  newConfig: AgentConfigurationIntake;
+  setNewConfig: React.Dispatch<React.SetStateAction<AgentConfigurationIntake>>;
+  onClickNext: () => void;
+}
+
+export function ServicePage({ newConfig, setNewConfig, onClickNext }: Props) {
+  const { data: serviceNames = [], status: serviceNamesStatus } = useFetcher(
+    callApmApi => {
+      return callApmApi({
+        pathname: '/api/apm/settings/agent-configuration/services',
+        forceCache: true
+      });
+    },
+    [],
+    { preservePreviousData: false }
+  );
+
+  const { data: environments = [], status: environmentStatus } = useFetcher(
+    callApmApi => {
+      if (newConfig.service.name) {
+        return callApmApi({
+          pathname: '/api/apm/settings/agent-configuration/environments',
+          params: {
+            query: { serviceName: omitAllOption(newConfig.service.name) }
+          }
+        });
+      }
+    },
+    [newConfig.service.name],
+    { preservePreviousData: false }
+  );
+
+  const { status: agentNameStatus } = useFetcher(
+    async callApmApi => {
+      const serviceName = newConfig.service.name;
+
+      if (!isString(serviceName) || serviceName.length === 0) {
+        return;
+      }
+
+      const { agentName } = await callApmApi({
+        pathname: '/api/apm/settings/agent-configuration/agent_name',
+        params: { query: { serviceName } }
+      });
+
+      setNewConfig(prev => ({ ...prev, agent_name: agentName }));
+    },
+    [newConfig.service.name, setNewConfig]
+  );
+
+  const ALREADY_CONFIGURED_TRANSLATED = i18n.translate(
+    'xpack.apm.agentConfig.servicePage.alreadyConfiguredOption',
+    { defaultMessage: 'already configured' }
+  );
+
+  const serviceNameOptions = serviceNames.map(name => ({
+    text: getOptionLabel(name),
+    value: name
+  }));
+  const environmentOptions = environments.map(
+    ({ name, alreadyConfigured }) => ({
+      disabled: alreadyConfigured,
+      text: `${getOptionLabel(name)} ${
+        alreadyConfigured ? `(${ALREADY_CONFIGURED_TRANSLATED})` : ''
+      }`,
+      value: name
+    })
+  );
+
+  return (
+    <EuiPanel paddingSize="m">
+      <EuiTitle size="xs">
+        <h3>
+          {i18n.translate('xpack.apm.agentConfig.servicePage.title', {
+            defaultMessage: 'Choose Service'
+          })}
+        </h3>
+      </EuiTitle>
+
+      <EuiSpacer size="m" />
+
+      {/* Service name options */}
+      <FormRowSelect
+        title="Service"
+        description={i18n.translate(
+          'xpack.apm.agentConfig.servicePage.serviceNameSelectHelpText',
+          { defaultMessage: 'Choose the service you want to configure.' }
+        )}
+        fieldLabel={i18n.translate(
+          'xpack.apm.agentConfig.servicePage.serviceNameSelectLabel',
+          { defaultMessage: 'Service Name' }
+        )}
+        isLoading={serviceNamesStatus === 'loading'}
```

Comment:
You could use `FETCH_STATUS` enum from useFetcher.
```suggestion
        isLoading={serviceNamesStatus === FETCH_STATUS.LOADING}
```